depmapomics/copynumbers.py

```diff
@@ -682,50 +874,76 @@ def postProcess(
-    print("summary of the gene cn data:")
-    print(genecn.values.min(), genecn.values.mean(), genecn.values.max())
-    failed = mut.checkAmountOfSegments(segments, thresh=segmentsthresh)
-
-    print("failed our QC")
-    print(failed)
-
-    if source_rename:
-        segments = segments.replace({"Source": source_rename})
-    if save_output:
-        h.listToFile(failed, save_output + "failed.txt")
-    # subsetting
-    segments = segments[
-        ~segments[constants.SAMPLEID].isin((set(failed) | set(todrop)) - set(priority))
-    ].reset_index(drop=True)
-    genecn = genecn[~genecn.index.isin((set(failed) | set(todrop)) - set(priority))]
+    mybiomart = mybiomart.drop_duplicates("ensembl_gene_id", keep="first")
 
-    # saving
-    print("saving files")
-    segments.to_csv(save_output + "segments_all.csv", index=False)
-    genecn.to_csv(save_output + "genecn_all.csv")
-    print("done")
+    if run_gatk_relative:
+        segments = loadFromGATKAggregation(
+            refworkspace,
+            setEntity=setEntity,
+            sampleset=sampleset,
+            sortby=sortby,
+            todrop=todrop,
+            doCleanup=doCleanup,
+        )
+        print("making gene level copy number")
+        # drop Ychrom if > maxYchrom
+        ychrom = segments[segments.Chromosome.str.contains("Y")]
+        countYdrop = [
+            i
+            for i in set(ychrom[constants.SAMPLEID])
+            if len(ychrom[ychrom[constants.SAMPLEID] == i]) > maxYchrom
+        ]
+        segments = segments[
+            ~(
+                (segments[constants.SAMPLEID].isin(countYdrop))
+                & (segments.Chromosome == "Y")
+            )
+        ]
+        genecn = mut.toGeneMatrix(
+            mut.manageGapsInSegments(segments),
+            mybiomart,
+            value_colname="SegmentMean",
+            gene_names_col="ensembl_gene_id",
+        )
+        # validation step
+        print("summary of the gene cn data:")
+        print(genecn.values.min(), genecn.values.mean(), genecn.values.max())
+        failed = mut.checkAmountOfSegments(segments, thresh=segmentsthresh)
+
+        print("failed our QC")
+        print(failed)
+
+        if source_rename:
+            segments = segments.replace({"Source": source_rename})
+        if save_output:
+            h.listToFile(failed, save_output + "failed.txt")
+        # subsetting
+        segments = segments[
+            ~segments[constants.SAMPLEID].isin(
+                (set(failed) | set(todrop)) - set(priority)
+            )
+        ].reset_index(drop=True)
+        genecn = genecn[~genecn.index.isin((set(failed) | set(todrop)) - set(priority))]
+
+        # saving
+        print("saving files")
+        segments.to_csv(save_output + "segments_all.csv", index=False)
+        genecn.to_csv(save_output + "genecn_all.csv")
+        print("done")
+
+    else:
+        segments, genecn = aggregate_cnvs_from_hmm(refworkspace)
+
+        print("saving files")
```

Comment:
Do we need to do this?

Reply:
Not really. At some point we had this idea that everything CDS/depmap release-specific should live in dm_omics.py, and functions in copynumbers.py and expression.py etc are more generic so public users can run them, and that's why the files are saved here. But I think the line is much more blurry now...